nodes/config/ui_base.html

```diff
@@ -657,30 +825,90 @@
 
         divTabs.append('<div class="nrdb2-layout-helptext">' + c_('label.layoutMessage') + '</div>')
 
+        /** @type {DashboardItemLookup} */
         const pages = {}
+        /** @type {DashboardItemLookup} */
         const groupsByPage = {}
+        /** @type {DashboardItemLookup} */
         const widgetsByGroup = {}
-
+        const widgetsBySubflowDef = {}
+        const subflowDefsWithWidgets = {}
+        const widgetsBelongingToSubflow = new Map()
+        const subflowInstances = {}
+        const subflowDefinitions = new Map()
+        const subflowDefIds = new Set()
         // get all pages & all groups
         RED.nodes.eachConfig(function (n) {
             if (n.type === 'ui-page' && !!n.ui) {
-                pages[n.id] = n
+                pages[n.id] = toDashboardItem(n)
             }
             if (n.type === 'ui-group' && !!n.page) {
                 if (!groupsByPage[n.page]) {
                     groupsByPage[n.page] = []
                 }
-                groupsByPage[n.page].push(n)
+                groupsByPage[n.page].push(toDashboardItem(n))
             }
         })
 
         // get all widgets
+        const uiNodesWithGroupProp = []
         RED.nodes.eachNode(function (n) {
             if (/^ui-/.test(n.type) && n.group) {
+                uiNodesWithGroupProp.push(n)
                 if (!widgetsByGroup[n.group]) {
                     widgetsByGroup[n.group] = []
                 }
-                widgetsByGroup[n.group].push(n)
+                widgetsByGroup[n.group].push(toDashboardItem(n))
+                if (n.z) {
+                    const subflowDef = RED.nodes.subflow(n.z)
+                    if (subflowDef) {
+                        subflowDefIds.add(subflowDef.id)
+                        widgetsBelongingToSubflow.set(n.id, subflowDef)
+                        subflowDefinitions.set('subflow:' + subflowDef.id, subflowDef)
+                        // get the env items that have type : "conf-types" and value : "ui-group"
+                        if (!widgetsBySubflowDef[subflowDef.id]) {
+                            widgetsBySubflowDef[subflowDef.id] = []
+                        }
+                        widgetsBySubflowDef[subflowDef.id].push(n)
+                        subflowDefsWithWidgets[subflowDef.id] = subflowDef
+                    }
+                }
+            }
+        })
+        // finally, get all subflow instances where the subflow has an an env var of `ui-group`
+        // const subflowDefIds = Object.keys(widgetsBySubflowDef).map(id => 'subflow:' + id)
```

Comment:
commented out code?

Reply:
tidied up in next commit - my apologies (and well spotted)

Reply:
resolved in [fef3ec1](https://github.com/FlowFuse/node-red-dashboard/pull/634/commits/fef3ec164c8042ce9e15e24550d7eb4faf84f0fb)